packages/wrangler/src/__tests__/versions/versions.deploy.test.ts

```diff
@@ -0,0 +1,900 @@
+			├ 1 Worker Version(s) selected
+			│
+			├     Worker Version 1:  00000000-0000-0000-0000-000000000000
+			│              Created:  1/4/2021, TIMESTAMP AM
+			│                  Tag:  -
+			│              Message:  -
+			│
+			├ What percentage of traffic should Worker Version 1 receive?
+			├ 100% of traffic
+			├
+			├ Add a deployment message (skipped)
+			│
+			│
+			│
+			╰  SUCCESS  Deployed named-worker version 00000000-0000-0000-0000-000000000000 at 100% (TIMINGS)"
+		`);
+		});
+
+		test("fails without --name arg", async () => {
+			const result = runWrangler(
+				"versions deploy 10000000-0000-0000-0000-000000000000 --yes --experimental-gradual-rollouts"
+			);
+
+			await expect(result).rejects.toMatchInlineSnapshot(
+				`[Error: You need to provide a name when deploying a worker. Either pass it as a cli arg with \`--name <name>\` or in your config file as \`name = "<name>"\`]`
+			);
+		});
+	});
+
+	describe("with wrangler.toml", () => {
+		beforeEach(writeWranglerToml);
+
+		test("no args", async () => {
+			const result = runWrangler(
+				"versions deploy --yes --experimental-gradual-rollouts"
+			);
+
+			await expect(result).rejects.toMatchInlineSnapshot(
+				`[Error: You must select at least 1 version to deploy.]`
+			);
+
+			expect(normalizeOutput(std.out)).toMatchInlineSnapshot(`
+			"╭ Deploy Worker Versions by splitting traffic between multiple versions
+			│
+			│
+			├ Your current deployment has 2 version(s):
+			│
+			│ (10%) 00000000-0000-0000-0000-000000000000
+			│       Created:  1/4/2021, TIMESTAMP AM
+			│           Tag:  -
+			│       Message:  -
+			│
+			│ (90%) 00000000-0000-0000-0000-000000000000
+			│       Created:  2/3/2021, TIMESTAMP AM
+			│           Tag:  -
+			│       Message:  -
+			│
+			│
+			├ Which version(s) do you want to deploy?
+			├ 0 Worker Version(s) selected
+			│"
+		`);
+		});
+
+		test("1 version @ (implicit) 100%", async () => {
+			const result = runWrangler(
+				"versions deploy 10000000-0000-0000-0000-000000000000 --yes --experimental-gradual-rollouts"
+			);
+
+			await expect(result).resolves.toBeUndefined();
+
+			expect(normalizeOutput(std.out)).toMatchInlineSnapshot(`
+			"╭ Deploy Worker Versions by splitting traffic between multiple versions
+			│
+			│
+			├ Your current deployment has 2 version(s):
+			│
+			│ (10%) 00000000-0000-0000-0000-000000000000
+			│       Created:  1/4/2021, TIMESTAMP AM
+			│           Tag:  -
+			│       Message:  -
+			│
+			│ (90%) 00000000-0000-0000-0000-000000000000
+			│       Created:  2/3/2021, TIMESTAMP AM
+			│           Tag:  -
+			│       Message:  -
+			│
+			│
+			├ Which version(s) do you want to deploy?
+			├ 1 Worker Version(s) selected
+			│
+			├     Worker Version 1:  00000000-0000-0000-0000-000000000000
+			│              Created:  1/4/2021, TIMESTAMP AM
+			│                  Tag:  -
+			│              Message:  -
+			│
+			├ What percentage of traffic should Worker Version 1 receive?
+			├ 100% of traffic
+			├
```

Comment:
I didn't want to mess too much with the CLI package because of existing usage – I'm not doing anything special with it here so it can and should be fixed there but can be done as a follow-up when thought through properly.

For now, this is a minor design glitch I think we're fine with going ahead with